File: ios/Classes/RNKLineViewCommands.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNKLineViewCommands, NSObject)

RCT_EXTERN_METHOD(setData:(nonnull NSNumber *)reactTag candles:(NSArray *)candles)
RCT_EXTERN_METHOD(appendCandle:(nonnull NSNumber *)reactTag candle:(NSDictionary *)candle)
RCT_EXTERN_METHOD(updateLastCandle:(nonnull NSNumber *)reactTag candle:(NSDictionary *)candle)
RCT_EXTERN_METHOD(unPredictionSelect:(nonnull NSNumber *)reactTag unused:(id)unused)

@end

File: ios/Classes/RNKLineViewCommands.swift
//
//  RNKLineViewCommands.swift
//  react-native-kline-view
//

import Foundation
import React

@objc(RNKLineViewCommands)
@objcMembers
class RNKLineViewCommands: NSObject, RCTBridgeModule {
    static func moduleName() -> String! {
        return "RNKLineViewCommands"
    }

    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    @objc var bridge: RCTBridge!

    @objc func setData(_ reactTag: NSNumber, candles: NSArray) {
        bridge.uiManager.addUIBlock { (_, viewRegistry) in
            guard let view = viewRegistry?[reactTag] as? HTKLineContainerView else { return }
            let list = candles as? [[String: Any]] ?? []
            view.setData(list)
        }
    }

    @objc func appendCandle(_ reactTag: NSNumber, candle: NSDictionary) {
        bridge.uiManager.addUIBlock { (_, viewRegistry) in
            guard let view = viewRegistry?[reactTag] as? HTKLineContainerView else { return }
            let dict = candle as? [String: Any] ?? [:]
            view.appendCandle(dict)
        }
    }

    @objc func updateLastCandle(_ reactTag: NSNumber, candle: NSDictionary) {
        bridge.uiManager.addUIBlock { (_, viewRegistry) in
            guard let view = viewRegistry?[reactTag] as? HTKLineContainerView else { return }
            view.updateLastCandle(dict)
        }
    }

    @objc func unPredictionSelect(_ reactTag: NSNumber, _ unused: Any?) {
        bridge.uiManager.addUIBlock { (_, viewRegistry) in
            guard let view = viewRegistry?[reactTag] as? HTKLineContainerView else { return }
            view.unPredictionSelect()
        }
    }
}

